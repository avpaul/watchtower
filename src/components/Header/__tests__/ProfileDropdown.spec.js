import React from 'react';
import { shallow } from 'enzyme';
import ProfileDropdown from '../ProfileDropDown';

describe('Profile Dropdown', () => {
  const setup = role => {
    const wrapper = shallow(
      <ProfileDropdown
        user={{
          name: 'Test User',
          picture: 'http://',
          roles: {
            [role]: 'ID'
          }
        }}
      />
    );

    return { wrapper };
  };

  const testProfileDropDownUsingRole = (role, roleName) => {
    const { wrapper } = setup(role);
    expect(wrapper.find('.profile-details__role').text()).toBe(roleName);
  };

  it('renders to match snapshot', () => {
    const { wrapper } = setup('Fellow');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected details for a WATCH_TOWER_EM', () =>
    testProfileDropDownUsingRole('WATCH_TOWER_EM', 'Engineering Manager'));

  it('renders the expected details for a WATCH_TOWER_SL', () =>
    testProfileDropDownUsingRole('WATCH_TOWER_SL', 'Simulations Lead'));

  it('renders the expected details for a WATCH_TOWER_LF', () =>
    testProfileDropDownUsingRole('WATCH_TOWER_LF', 'Learning Facilitator'));

  it('renders the expected details for a WATCH_TOWER_TTL', () =>
    testProfileDropDownUsingRole('WATCH_TOWER_TTL', 'Technical Team Lead'));

  it('renders the expected details for a WATCH_TOWER_OPS', () =>
    testProfileDropDownUsingRole('WATCH_TOWER_OPS', 'Operations Manager'));

  it('renders the expected details when incorrect role has been provided', () =>
    testProfileDropDownUsingRole('Default', 'Fellow'));

  it('renders an empty div if the user object does not contain data', () => {
    const wrapper = shallow(<ProfileDropdown user={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
