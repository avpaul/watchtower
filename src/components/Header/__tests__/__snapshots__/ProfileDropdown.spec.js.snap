// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Dropdown renders an empty div if the user object does not contain data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileDropDown
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dropdown-item disabled pl-0 pr-0"
>
          <div />
</div>,
        <a
          className="dropdown-item"
          data-target="#logout-modal"
          data-toggle="modal"
          href="/"
>
          Log out
</a>,
      ],
      "className": "dropdown-menu dropdown-menu-right",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "dropdown-item disabled pl-0 pr-0",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log out",
          "className": "dropdown-item",
          "data-target": "#logout-modal",
          "data-toggle": "modal",
          "href": "/",
        },
        "ref": null,
        "rendered": "Log out",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dropdown-item disabled pl-0 pr-0"
>
            <div />
</div>,
          <a
            className="dropdown-item"
            data-target="#logout-modal"
            data-toggle="modal"
            href="/"
>
            Log out
</a>,
        ],
        "className": "dropdown-menu dropdown-menu-right",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "className": "dropdown-item disabled pl-0 pr-0",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log out",
            "className": "dropdown-item",
            "data-target": "#logout-modal",
            "data-toggle": "modal",
            "href": "/",
          },
          "ref": null,
          "rendered": "Log out",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Profile Dropdown renders to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileDropDown
    user={
        Object {
            "name": "Test User",
            "picture": "http://",
            "roles": Object {
              "Fellow": "ID",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dropdown-item disabled pl-0 pr-0"
>
          <div
                    className="profile-details"
          >
                    <div
                              className="profile-details__image"
                    >
                              <img
                                        alt="Test User profile-details pic"
                                        src="http://"
                              />
                    </div>
                    <div
                              className="profile-details__right"
                    >
                              <div
                                        className="profile-details__name"
                              >
                                        Test User
                              </div>
                              <div
                                        className="profile-details__role"
                              >
                                        Fellow
                              </div>
                    </div>
          </div>
</div>,
        <a
          className="dropdown-item"
          data-target="#logout-modal"
          data-toggle="modal"
          href="/"
>
          Log out
</a>,
      ],
      "className": "dropdown-menu dropdown-menu-right",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="profile-details"
>
            <div
                        className="profile-details__image"
            >
                        <img
                                    alt="Test User profile-details pic"
                                    src="http://"
                        />
            </div>
            <div
                        className="profile-details__right"
            >
                        <div
                                    className="profile-details__name"
                        >
                                    Test User
                        </div>
                        <div
                                    className="profile-details__role"
                        >
                                    Fellow
                        </div>
            </div>
</div>,
          "className": "dropdown-item disabled pl-0 pr-0",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="profile-details__image"
>
                <img
                                alt="Test User profile-details pic"
                                src="http://"
                />
</div>,
              <div
                className="profile-details__right"
>
                <div
                                className="profile-details__name"
                >
                                Test User
                </div>
                <div
                                className="profile-details__role"
                >
                                Fellow
                </div>
</div>,
            ],
            "className": "profile-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Test User profile-details pic"
                  src="http://"
/>,
                "className": "profile-details__image",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Test User profile-details pic",
                  "src": "http://",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="profile-details__name"
>
                    Test User
</div>,
                  <div
                    className="profile-details__role"
>
                    Fellow
</div>,
                ],
                "className": "profile-details__right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Test User",
                    "className": "profile-details__name",
                  },
                  "ref": null,
                  "rendered": "Test User",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Fellow",
                    "className": "profile-details__role",
                  },
                  "ref": null,
                  "rendered": "Fellow",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log out",
          "className": "dropdown-item",
          "data-target": "#logout-modal",
          "data-toggle": "modal",
          "href": "/",
        },
        "ref": null,
        "rendered": "Log out",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dropdown-item disabled pl-0 pr-0"
>
            <div
                        className="profile-details"
            >
                        <div
                                    className="profile-details__image"
                        >
                                    <img
                                                alt="Test User profile-details pic"
                                                src="http://"
                                    />
                        </div>
                        <div
                                    className="profile-details__right"
                        >
                                    <div
                                                className="profile-details__name"
                                    >
                                                Test User
                                    </div>
                                    <div
                                                className="profile-details__role"
                                    >
                                                Fellow
                                    </div>
                        </div>
            </div>
</div>,
          <a
            className="dropdown-item"
            data-target="#logout-modal"
            data-toggle="modal"
            href="/"
>
            Log out
</a>,
        ],
        "className": "dropdown-menu dropdown-menu-right",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="profile-details"
>
              <div
                            className="profile-details__image"
              >
                            <img
                                          alt="Test User profile-details pic"
                                          src="http://"
                            />
              </div>
              <div
                            className="profile-details__right"
              >
                            <div
                                          className="profile-details__name"
                            >
                                          Test User
                            </div>
                            <div
                                          className="profile-details__role"
                            >
                                          Fellow
                            </div>
              </div>
</div>,
            "className": "dropdown-item disabled pl-0 pr-0",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="profile-details__image"
>
                  <img
                                    alt="Test User profile-details pic"
                                    src="http://"
                  />
</div>,
                <div
                  className="profile-details__right"
>
                  <div
                                    className="profile-details__name"
                  >
                                    Test User
                  </div>
                  <div
                                    className="profile-details__role"
                  >
                                    Fellow
                  </div>
</div>,
              ],
              "className": "profile-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Test User profile-details pic"
                    src="http://"
/>,
                  "className": "profile-details__image",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Test User profile-details pic",
                    "src": "http://",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="profile-details__name"
>
                      Test User
</div>,
                    <div
                      className="profile-details__role"
>
                      Fellow
</div>,
                  ],
                  "className": "profile-details__right",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Test User",
                      "className": "profile-details__name",
                    },
                    "ref": null,
                    "rendered": "Test User",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Fellow",
                      "className": "profile-details__role",
                    },
                    "ref": null,
                    "rendered": "Fellow",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log out",
            "className": "dropdown-item",
            "data-target": "#logout-modal",
            "data-toggle": "modal",
            "href": "/",
          },
          "ref": null,
          "rendered": "Log out",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
